### DISCUSSION TITLE
更换框架到Lagrange.Core后，出现“部分图片无法获取，请尝试使用其他设备QQ发送”

### DISCUSSION THREAD

**Discussion by WenHe233** [Tsuk1ko/CQ-picfinder-robot#454] [问题求助] [answered]
如题，之前使用的框架是[OpenShamrock](https://github.com/whitechi73/OpenShamrock)，在使用时没有出现问题。最近试着迁移到了[Lagrange.Core](https://github.com/LagrangeDev/Lagrange.Core)，却发现在使用这个插件时出现了问题。

连接后试图搜图，一直显示`部分图片无法获取，请尝试使用其他设备QQ发送`。

尝试过更换设备和使用其他图片，出现了相同的情况。
试着在搜图时添加了`--debug`参数，也还是显示`部分图片无法获取，请尝试使用其他设备QQ发送`。
![image](https://github.com/Tsuk1ko/cq-picsearcher-bot/assets/43093562/a71abbae-7d58-42d8-a565-768fa1c031cf)

Lagrange.Core配置文件的正向ws部分：
```
        {
            "Type": "ForwardWebSocket",
            "Host": "0.0.0.0",
            "Port": 6700,
            "HeartBeatInterval": 5000,
            "AccessToken": ""
          },
```
插件的cqws部分：
```
  "cqws": {
    "host": "192.168.31.156",
    "port": 6700,
    "enableAPI": true,
    "enableEvent": true,
    "accessToken": "",
    "reconnection": true,
    "reconnectionAttempts": 10,
    "reconnectionDelay": 5000
  },
```

想问问有没有使用Lagrange.Core的朋友遇到过类似的情况，后面是如何解决的

**Comment by Tsuk1ko** [comment]
🤔这个提示实际来源于对以前一个客户端图片 url bug 的判断，不排除有误判可能

Lagrange.Core 是否跟 go-cqhttp 一样有消息内容日志，提供一下 cq 码中的 url

或者 pull 一下再复现一次，我增加了 debug 模式下的日志输出

**Comment by WenHe233** [comment]
Lagrange.Core中的日志：
```
trce: Lagrange.OneBot.Core.Network.Service.ForwardWSService[3]
      Send(Conn: ): {"message_type":"private","sub_type":"friend","message_id":*****,"user_id":*****,"message":[{"type":"image","data":{"file":"http://gchat.qpic.cn/offpic_new/2124060098//2124060098-1531467184-64D35AF69601BA98852152A2F06D3B71/0?term=255\u0026is_origin=1","url":"http://gchat.qpic.cn/offpic_new/2124060098//2124060098-1531467184-64D35AF69601BA98852152A2F06D3B71/0?term=255\u0026is_origin=1"}}],"raw_message":"[CQ:image,file=http://gchat.qpic.cn/offpic_new/2124060098//2124060098-1531467184-64D35AF69601BA98852152A2F06D3B71/0?term=255\u0026amp;is_origin=1]","font":0,"sender":{"user_id":*****,"nickname":"*****","sex":"unknown"},"time":1708252948,"self_id":*****,"post_type":"message"}
```
[CQ码中的URL](http://gchat.qpic.cn/offpic_new/2124060098//2124060098-1531467184-64D35AF69601BA98852152A2F06D3B71/0?term=255\u0026amp;is_origin=1)

~~我是用docker部署的，目前看镜像好像没更新（~~
尝试着本地重新拉取项目部署了一下，没有看到日志输出
![image](https://github.com/Tsuk1ko/cq-picsearcher-bot/assets/43093562/2a5646d8-d4d4-4193-9dcd-a816dcb42a4c)
![image](https://github.com/Tsuk1ko/cq-picsearcher-bot/assets/43093562/bc448983-0d86-400d-b866-d1bb2ae507e5)

**Reply by Tsuk1ko** [reply, marked as answer]
破案了，Lagrange.Core CQ:image 和 go-cqhttp 不太一样，file 里面是 url

```
[CQ:image,file=http://gchat.qpic.cn/offpic_new/2124060098//2124060098-1531467184-64D35AF69601BA98852152A2F06D3B71/0?term=255&amp;is_origin=1]
```

在新版本 3.16.3 兼容了
